pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -105,47 +104,41 @@ public class PinotQueryResource {
   @POST
   @Path("sql")
   @ManualAuthorization // performed by broker
-  public String handlePostSql(String requestJsonStr, @Context HttpHeaders httpHeaders) {
+  public StreamingOutput handlePostSql(String requestJsonStr, @Context HttpHeaders httpHeaders) {
```

Comment:
What's the purpose of changing the return type to `StreamingOutput`?

Reply:
The advantage of returning a StreamingOutput instead of a String is double:
1. We don't need to allocate the response on heap. Instead we can just pipeline the response from the broker. We use the same trick in LocalLogFileServer and TablesResource.
2. We can start to pipeline results to the customer as soon as the first bytes from broker arrive. Before we had to wait until the whole response was received in the controller.